lib/faraday/rack_builder.rb

```diff
@@ -84,6 +84,7 @@ def use(klass, *args, &block)
         use_symbol(Faraday::Middleware, klass, *args, &block)
       else
         raise_if_locked
+        raise_if_adapter_set unless is_adapter?(klass)
```

Comment:
Yep that's fine.
Leaving it there will cause multiple adapters to be inserted as you said, which is definitely something we don't want to happen 😃